### PR TITLE
replace TimGM6mb.sf2 with GeneralUser_GS_v1.511.sf3

### PR DESCRIPTION
**Pull request by rfomin**
It seems higher quality and better balanced. Thanks to @ceski-1 for recommendation. Also one of the recommended soundfonts for MuseScore: https://musescore.org/en/handbook/3/soundfonts-and-sfz-files#gm_soundfonts

Taken from here: https://github.com/dwhinham/mt32-pi/blob/f81aa797906deee415b957b42efa87a135cba7f1/.github/workflows/ci.yml#L192
I compressed it into .sf3 format (lossy OGG) to reduce the size.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -200,11 +200,9 @@ Copyright:
  © 2013-2018 Alexey Khokholov (Nuke.YKT).  
 License: [GPL-2.0+](https://www.gnu.org/licenses/old-licenses/gpl-2.0.html)
 
-Files: `soundfonts/TimGM6mb.sf2`  
+Files: `soundfonts/GeneralUser_GS_v1.51.sf3`
```
Comment (ceski-1) [inline README.md:203]: Did you mean `GeneralUser_GS_v1.511.sf3` ?

**Comment by ceski-1** [comment]
This has always seemed like a solid soundfont that works well with just about everything. This kind of topic is always pretty subjective, so I'm curious how it'll be received.

File: README.md (hunk 1 above)
Comment (rfomin) [reply README.md]: Yes, thanks. Fixed in https://github.com/fabiangreffrath/woof/pull/743/commits/edb57a544171867464eafeee22226b327059a456

File: README.md [hunk 2]
```diff
@@ -200,11 +200,9 @@ Copyright:
  © 2013-2018 Alexey Khokholov (Nuke.YKT).  
 License: [GPL-2.0+](https://www.gnu.org/licenses/old-licenses/gpl-2.0.html)
 
-Files: `soundfonts/TimGM6mb.sf2`  
+Files: `soundfonts/GeneralUser_GS_v1.511.sf3`  
 Copyright:  
- © 2004 Tim Brechbill;  
- © 2010 David Bolton.  
-License: [GPL-2.0](https://www.gnu.org/licenses/old-licenses/gpl-2.0.html)
+ © 2020 S. Christian Collins
```
Comment (fabiangreffrath) [inline README.md:205]: Please do at least point to the local copy of the custom (*sigh!*) license and call it "GeneralUser GS v1.51 License v2.0" or similar.

File: soundfonts/LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+Many of the samples are original, but some were taken from other banks freely (and legally) available on the Internet from various SoundFont websites.  Because GeneralUser GS originated as a personal project with no intention for publication, I cannot be 100% sure where all of the samples originated, although I do know that none of them came from commercially published SoundFont packages or sample CDs.  Regardless, many "free" SoundFonts available on the web may indeed contain samples of questionable origin.  My understanding of the copyrights of all samples is only as good as the information provided by the original sources. If you become aware of any restricted samples being used in GeneralUser GS, please let me know so I can replace them.
+
+This uncertainty may concern you if you intend to use GeneralUser GS in a commercial software product.  That being said, I have never received any complaint regarding sample ownership since I published the original GeneralUser GS back in 2000, and as far as I am aware, neither have any of the companies creating commercial software products using GeneralUser GS.
```
Comment (fabiangreffrath) [inline soundfonts/LICENSE.txt:12]: This is really very vague and in fact not better than any other license for any other replacement soundfont suggested so far. Sorry, but I just can't stand "I have taken some samples from the Internet, but nobody has complained so far" type of disclaimers. Not sure what to think about this...

Comment (rfomin) [reply soundfonts/LICENSE.txt:12]: All we have for TimGM6mb.sf2 is Tim Brechbill word: https://musescore.org/en/node/1463#comment-9168
I don't think a completely legal GM soundfont exist. If only Roland would release their sound font for public use...

File: README.md (hunk 2 above)
Comment (rfomin) [reply README.md:205]: How about "public domain"? In fact, it is.

File: soundfonts/LICENSE.txt (hunk 3 above)
Comment (fabiangreffrath) [reply soundfonts/LICENSE.txt:12]: Yes, but at least that *is* a word. And the choice for this being distributable under GPL is backed up by varios Linux distros packaging it as their default soundfont.

File: README.md (hunk 2 above)
Comment (fabiangreffrath) [reply README.md:205]: No, wait, public domain needs to be explicitly declared as such.

**Comment by rfomin** [comment]
Another alternative is MuseScore_General.sf3 from the same author, released under the MIT license https://musescore.org/en/handbook/3/soundfonts-and-sfz-files#gm_soundfonts.
There's even better quality and balance, but the file size is 35.9 MB in sf3 format and over 200 MB in memory, probably too much for a Doom ports.

**Comment by mikeday0** [comment]
I agree with NightFright2k19 about SF3 files. The extra load time every time the music changes can be annoying. Maybe if you have a more recent computer it's not so bad but I know there are a lot of users on old hardware.

**Comment by rfomin** [comment]
> The extra load time every time the music changes can be annoying.

The soundfont loaded only on startup, not on track change.

**Comment by mikeday0** [comment]
> > The extra load time every time the music changes can be annoying.
> 
> The soundfont loaded only on startup, not on track change.

It has been a while, but my previous experience with FluidSynth and sf3 files was that there were long loads every time the track changed. Maybe this has been changed in a more recent version.

**Comment by rfomin** [comment]
> It has been a while, but my previous experience with FluidSynth and sf3 files was that there were long loads every time the track changed. Maybe this has been changed in a more recent version.

GZDoom does this for a reason.

**Comment by mikeday0** [comment]
> Edit: Just checked, SDL_Mixer does this too. However, for Woof this is not a problem.

That explains it. I would have been using FluidSynth via SDL_Mixer.

**Comment by ceski-1** [comment]
> That and the aspect of lossy compression, even if the average user may not hear any difference.

This shouldn't be a problem with an appropriate quality setting. For example, q6 for ogg is a little better than 192kbps for mp3, which is already considered transparent. Even q5 is fine (> 160kbps mp3).

**Comment by rfomin** [comment]
> @rfomin do you know what setting was used?

I converted GeneralUser_GS_v1.511.sf3 using [polyphone](https://www.polyphone-soundfonts.com/documentation/en/manual/annexes/command-line) tool with default settings (medium - I don't know what that means exactly).

I downloaded MuseScore_General.sf3 so I don't know how it's converted, but it's impractical anyway (too big). Also hi quality samples sound out of place in Doom, IMHO.

**Comment by ceski-1** [comment]
> default settings (medium - I don't know what that means exactly).

[Medium quality is 0.6](https://github.com/davy7125/polyphone/blob/master/sources/core/output/sf3/outputsf3.cpp#L62), which means q6, so that's ~192kbps. Not bad!

**Comment by fabiangreffrath** [comment]
> I think it's important to understand the context too. This is supposed to be a solid default soundfont. Someone who wants the uncompressed version, or any other soundfont, can easily replace it.

Doesn't the same apply to the one we already have?

**Comment by rfomin** [comment]
> Doesn't the same apply to the one we already have?

Sure, but sometimes it sounds poor. I thought we should bundle a decent soundfont with Crispy, PrBoom+ and Woof.

**Comment by ceski-1** [comment]
> Doesn't the same apply to the one we already have?

Lossless compression, or no compression at all, is important when the context is archiving. You wouldn't want to lossy compress something that's already lossy compressed. But here we have a soundfont that's using lossy compression at a high enough quality setting that it's transparent.

**Comment by fabiangreffrath** [comment]
> > Doesn't the same apply to the one we already have?
> 
> Lossless compression, or no compression at all, is important when the context is archiving. You wouldn't want to lossy compress something that's already lossy compressed. But here we have a soundfont that's using lossy compression at a high enough quality setting that it's transparent.

I was merely referring to its role as the "solid default soundfont".

**Comment by ceski-1** [comment]
I think this thread perfectly sums up the continued longevity of TimGM6mb!

**Comment by NightFright2k19** [comment]
I don't want to insert myself too much into this conversation since I am not involved in the project, but soundfonts are such a controversial topic with everyone having their own favorite... It basically doesn't matter what you use as a default as long as it works. Most users will replace the default soundfont with something else, anyway. 

Your current default has small filesize, sounds OK and doesn't need compressed samples. For a starter solution, I'd say it's enough.

**Comment by rfomin** [comment]
> Most users will replace the default soundfont with something else, anyway (as you pointed out before).

I think most users use [gzdoom.sf2](https://github.com/ZDoom/gzdoom/blob/master/soundfont/gzdoom.sf2) without even knowing it. Now DSDA-Doom also uses it by default, I think it will become standard, like MS GS Wavetable Synth before. Unfortunately, it does not have a license.

**Comment by rfomin** [comment]
Unfortunately, the author of the soundfont "GeneralUser GS" cannot release it under a free license. I think we should wait for another alternative.

**Pull request closed without merging**